### PR TITLE
Model and application archives In persistent persistent volume

### PR DESCRIPTION
**Pull request by sankarpn**
Add tests to create domain from modelHome property and model file, application archives stored in persistent volume.

Jenkins Results

https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/2044/

### CODE REVIEW

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomainModelInPV.java [hunk 1]
```diff
@@ -0,0 +1,447 @@
...
+
+  /**
+   * 1. Get namespaces for operator and WebLogic domain.
+   * 2. Create operator.
+   * 3. Build a MII with no domain and push it
+   * to repository.
+   * 4. Create WebLogic credential and model encryption secrets
+   * 5. Create PV and PVC to store model and
+   * application files.
+   * 6. Copy the files to model and application files to PV.
```
[the remark this answers is not in the document]
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomainModelInPV.java:139]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomainModelInPV.java [hunk 2]
```diff
@@ -0,0 +1,447 @@
...
+              logger.info(managedServer + " is healthy");
+            } else {
+              logger.info(managedServer + " health is not OK or server not found");
+            }
+          }
+          return health;
+        });
+  }
+
+  // setup a temporary to access PV location to store model and application files
```
[the remark this answers is not in the document]
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomainModelInPV.java:355]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomainModelInPV.java [hunk 3]
```diff
@@ -0,0 +1,447 @@
...
+  }
+
+  /**
+   * Test domain creation from model file stored in PV.
+   * 1. Create the domain custom resource using mii with no domain and specifying a PV location for modelHome
+   * 2. Verify the domain creation is successful and application is accessible.
+   */
+  @Test
+  @DisplayName("Create MII domain with model and application file from PV")
+  public void testCreateDomainWithModelInPV() {
```
[the remark this answers is not in the document]
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomainModelInPV.java:231]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomainModelInPV.java [hunk 4]
```diff
@@ -0,0 +1,445 @@
...
+      managedServerNames.add(MANAGED_SERVER_NAME_BASE + i);
+    }
+
+    //verify admin server accessibility and the health of cluster members
+    verifyMemberHealth(adminServerPodName, managedServerNames, ADMIN_USERNAME_DEFAULT, ADMIN_PASSWORD_DEFAULT);
+
+  }
+
+  // create custom domain resource with model file in modelHome
+  private Domain buildDomainResource() {
```
Comment (doxiao) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomainModelInPV.java:246]: This method looks similar to CommonMiiTestUtils.createDomainResource(). Can we use that one (make it public first), and add additional things on top of the domain created by that method?

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomainModelInPV.java [hunk 5]
```diff
@@ -0,0 +1,445 @@
...
+    }
+    try {
+      logger.info("Creating directory {0} in PV", "/shared/model");
+      Exec.exec(webLogicPod, null, false, "/bin/sh", "-c", "mkdir /shared/model");
+    } catch (IOException | ApiException | InterruptedException ex) {
+      logger.warning(ex.getMessage());
+    }
+
+    try {
+      //copy the model file to PV using the temp pod - we don't have access to PVROOT in Jenkins env
```
[the remark this answers is not in the document]
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomainModelInPV.java:203]: Yes that is ideal. But in our Jenkins setup we don't have access to the PVROOT on Kubelet nodes and we  cannot directly copy the files to hostPath and mount it.

[the remark this answers is not in the document]
Comment (doxiao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomainModelInPV.java:191]: okay. thanks.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomainModelInPV.java (hunk 4 above)
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomainModelInPV.java:234]: Using the method from CommonMiiTestUtils.createDomainResource()

**Review by anpanigr — APPROVED** [review]
LGTM

**Review by doxiao — APPROVED** [review]
LGTM

**Pull request merged**